src/sinks/datadog/logs/tests.rs

```diff
@@ -496,3 +498,17 @@ async fn error_is_retriable() {
     //       but are not straightforward to instantiate due to the design of
     //       the crates they originate from.
 }
+
+#[tokio::test]
+async fn cannot_configure_restricted_headers() {
```

Comment:
It would be nice to document the headers that cannot be set though I realize that will involve some cue changes.

Reply:
[95c7ccb](https://github.com/vectordotdev/vector/pull/15745/commits/95c7ccbbb5a2d608cff35143158a9a763822f25e) Updated to allow overwriting any header